.github/workflows/build.yml

```diff
@@ -0,0 +1,20 @@
+name: Build
+
+on: push
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - run: git submodule update --init --recursive
+      - name: Build project and run tests
+        run: ./gradlew build --stacktrace
+      - name: Upload code coverage report
+        uses: codecov/codecov-action@v1
```

Comment:
Codecov also has the v2 action. Also, v1 is up for deprecation and sunset in early 2022 which is pretty soon.